alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -494,6 +541,31 @@ class TransportIndexMonitorAction @Inject constructor(
             if (request.monitor.enabled && currentMonitor.enabled)
                 request.monitor = request.monitor.copy(enabledTime = currentMonitor.enabledTime)
 
+            // On update monitor check which backend roles to associate to the monitor
+            if (user != null) {
+                if (request.rbacRoles != null) {
+                    if (isAdmin(user)) {
+                        request.monitor = request.monitor.copy(
+                            user = User(user.name, request.rbacRoles, user.roles, user.customAttNames)
+                        )
+                    } else {
+                        val rolesToRemove = user.backendRoles
+                        request.rbacRoles.orEmpty().let { rolesToRemove.removeAll(it) }
+                        val rolesToAdd = request.rbacRoles?.intersect(user.backendRoles)
+                        val updatedRbac = currentMonitor.user?.backendRoles
+                        updatedRbac?.removeAll(rolesToRemove)
+                        rolesToAdd.orEmpty().let { updatedRbac?.addAll(it) }
```

Comment:
This is a little hard to read. Am I understand right that we're essentially:

1. Removing the backend roles (currently on the Monitor) which are associated with the calling user **but not** mentioned in the request
2. We're adding the backend roles that are both in the request **and** the calling user has associated as well

So, for example, say we have a Monitor with backend roles `[a, b, c, d]` associated with it.

If I'm User A and I have backend roles `[a, b, c]` associated with me and I make a request to update the Monitor's backend roles to `[a, b]`. This would mean that:

1. The roles to remove are `[c]`
2. The roles to add are `[a, b]`

I'm assuming this is so that the role `[d]` which I never had associated to me, doesn't get removed when I update the Monitor? Is this the behavior that we want? The `admin` wouldn't necessarily have all backend roles associated with them, wouldn't that mean they can't definitely reduce the associated roles in that case?

Reply:
We want the behavior such that role `[d]` is still there.

Reply:
added comments in code for this